tests/components/touchline_sl/test_config_flow.py

```diff
@@ -0,0 +1,137 @@
+"""Test the Roth Touchline SL config flow."""
+
+from typing import NamedTuple
+from unittest.mock import AsyncMock, patch
+
+from pytouchlinesl.client import RothAPIError
+
+from homeassistant import config_entries
+from homeassistant.components.touchline_sl.const import DOMAIN
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+
+class FakeModule(NamedTuple):
+    """Fake Module used for unit testing only."""
+
+    name: str
+    id: str
+
+
+FAKE_MODULES = [FakeModule(name="Foobar", id="deadbeef")]
+
+
+async def test_form(hass: HomeAssistant, mock_setup_entry: AsyncMock) -> None:
+    """Test we get the form."""
+    result1 = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result1["type"] == FlowResultType.FORM
+    assert result1["step_id"] == "user"
+    assert result1["errors"] == {}
+
+    with (
+        patch("pytouchlinesl.TouchlineSL.modules", return_value=FAKE_MODULES),
+        patch("pytouchlinesl.TouchlineSL.user_id", return_value=12345),
```

Comment:
Strongly recommend create a mock that completely mocks out the library class. Check Airgradient for an example